### ISSUE TITLE
Same thrift file name in different directories messes up generation.

### ISSUE THREAD

**Issue by Scrier** [morimekta/providence#77]
I have the following setup of a directory three under resources:

- thrift
  - folder_a
    - api.thrift
  - folder_b
    - api.thrift

But when I generate b it will take the namespace of a and will be unable to compile in the service method. 

a.api.thrift:
```
namespace java   i.s.p.c.a

struct ARequest {
    1: optional string name
}

struct AResponse {
    1: optional i32 name
}

service AService {

    AResponse aCall(1: ARequest request)

}
```

b.api.thrift:
```
namespace java   i.s.p.r.a

struct BRequest {
    1: optional string title;
}

struct BResponse {
    1: optional string title;
}

service BService {

    BResponse bCall(1: BRequest request)

}

```

The prefix in BService becomes:
```

    public interface Iface {
        i.s.p.c.a.BResponse bCall(
                i.s.p.c.a.BRequest pRequest)
    }
```

so the namespace from A instead of B.

Alot of removal of the code I copied from but can show you.

One thing might be that this is not supported by the thrift at all.

**Comment by morimekta** [comment]
Should be fixed in HEAD now. Next release should be soon.

**Comment by morimekta** [comment]
Fixed in 0.5.0
